postStart.sh

```diff
@@ -56,30 +51,37 @@ function setPasswordFromFile() {
 
 if curl --fail --silent -u "${USERNAME}":"${PASSWORD}" http://"${HOST}"/service/metrics/ping; then
     echo "Login to nexus succeeded. Default password worked. Updating password if available..."
-    setScriptList
-    createOrUpdateAndRun "${NEXUS_BASE_DIR}"/admin_password
+    curl --fail -X PUT -u "${USERNAME}":"${PASSWORD}" "http://${HOST}/service/rest/beta/security/users/admin/change-password" --header "Content-Type: text/plain" -d "${PASSWORD_FROM_FILE}"
     setPasswordFromFile
 elif [ -n "${PASSWORD_FROM_FILE}" ]; then
     setPasswordFromFile
     echo "Default password failed. Checking password file..."
     if curl --fail --silent -u "${USERNAME}":"${PASSWORD}" http://"${HOST}"/service/metrics/ping; then
         echo "Login to nexus succeeded. Password from secret file worked."
-        setScriptList
     else
         die "Login to nexus failed. Tried both the default password and the provided password secret file."
     fi
 else
     die "Login to nexus failed. Tried the default password only since no password secret file was provided."
 fi
 
-
-mapfile -t REPOS < <(find "${NEXUS_REPO_DIR}" -maxdepth 1 -type f -name "*json*" | sed -e 's/\..*$//')
+# For each maven repository proxy json file, create the repo proxy via the Nexus API.
+echo "Creating maven proxy repositories from json"
+mapfile -t REPOS < <(find "${NEXUS_MAVEN_REPO_DIR}" -maxdepth 1 -type f -name "*json*")
```

Comment:
`mapfile` is imo an unfortunate choice here imo, since I don't think it is generally available. Makes it for example hard to test this script locally.

Reply:
Agreed.  It feels okay to me to leave it since it was there before.  I installed a newer version of bash locally via brew and call it explicitly when testing.